rfcs/none-and-unknown-types.md

```diff
@@ -0,0 +1,99 @@
+# Add never and unknown types
+
+## Summary
+
+Add `unknown` and `never` types that are inhabited by everything and nothing respectively.
+
+## Motivation
+
+There are lots of cases in local type inference, semantic subtyping,
+and type normalization, where it would be useful to have top and
+bottom types. Currently, `any` is filling that role, but it has
+special "switch off the type system" superpowers.
+
+The `never` type comes up whenever type inference infers incompatible types for a variable, for example
+
+```lua
+  function oops(x)
+    print("hi " ++ x)  -- constrains x must be a string
```

Comment:
Is this a typo for `..`, or is this a new syntax proposed by some other RFC?

Reply:
Typo.